website/docs/extensions/write/onboarding-workflow.md

```diff
@@ -0,0 +1,531 @@
+
+![img1](img/setup_button.png)
+
+The enablement clause is mandatory and must be written by using [when clauses](when-clause-context.md).
+
+In the following example, we specify that the onboarding needs to be enabled if and only if the user's OS is Linux, and the `podmanIsNotInstalled` context value is true. Alternatively, if the user's OS is different from Linux, that the `podmanMachineExists` context value must be false. In other words, if the user is on Linux, the onboarding must be enabled only if podman is not installed; for all other operating systems, it should be enabled if there is no Podman machine.
+
+```json
+"enablement": "(isLinux && onboardingContext:podmanIsNotInstalled) || (!isLinux && !onboardingContext:podmanMachineExists)"
+```
+
+### Steps
+
+The steps property is required and includes the actual content that will be displayed to the user during the workflow.
+
+Let's look again at its schema
+
+```json
+"type": "object",
+"properties": {
+    "id": {
+        "type": "string"
+    },
+    "title": {
+        "type": "string"
+    },
+    "description": {
+        "type": "string"
+    },
+    "media": {
+        "path": {
+            "type": "string"
+        },
+        "altText": {
+            "type": "string"
+        },
+    },
+    "command": {
+        "type": "string"
+    },
+    "completionEvents": {
+        "type": "array",
+        "items": {
+            "type": "string"
+        }
+    },
+    "content": {
+        "type": "array",
+        "items": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "value": {
+                        "type": "string"
+                    },
+                    "highlight": {
+                        "type": "boolean"
+                    },
+                    "when": {
+                        "type": "string"
+                    }
+                },
+                "required": ["value"]
+            }
+        }
+    },
+    "when": {
+        "type": "string"
+    },
+    "component": {
+        "type": "string",
+        "enum": ["createContainerProviderConnection", "createKubernetesProviderConnection"]
+    },
+    "state": {
+        "type": "string",
+        "enum": ["completed", "failed"]
+    }
+},
+"required": ["id", "title"]
+```
+
+A step has only two mandatory fields - id and title. All other properties are optional.
+
+#### Id
+
+The **id** must be unique to identify a step, and it is never displayed directly to the user.
+
+#### Title, description and media
+
+The **title**, **description** and **media** works as explained earlier. The only difference is their placement - they will appear in the top-center of the body.
+
+![img2](img/step_title_media_description.png)
+
+**Note:** If media is not specified, Podman Desktop will display the extension's icon providing the onboarding.
```

Comment:
'extension's icon providing the onboarding' -> 'icon of the extension providing the onboarding'